lib/compat/wordpress-6.5/navigation-block-hooks.php

```diff
@@ -0,0 +1,119 @@
+	? $hooked_blocks[ $anchor_block_type ][ $last_child_position ]
+	: array();
+
+	$hooked_block_types_first_child = isset( $hooked_blocks[ $anchor_block_type ][ $first_child_position ] )
+	? $hooked_blocks[ $anchor_block_type ][ $first_child_position ]
+	: array();
+
+	$hooked_block_types_last_child  = apply_filters( 'hooked_block_types', $hooked_block_types_last_child, $last_child_position, $anchor_block_type, null );
+	$hooked_block_types_first_child = apply_filters( 'hooked_block_types', $hooked_block_types_first_child, $first_child_position, $anchor_block_type, null );
+
+	$ignored_hooked_blocks = array_merge( $hooked_block_types_last_child, $hooked_block_types_first_child );
+
+	if ( ! empty( $ignored_hooked_blocks ) ) {
+		update_post_meta( $post->ID, '_wp_ignored_hooked_blocks', json_encode( $ignored_hooked_blocks ) );
+	}
+}
+
+/**
+ * Hooks into the REST API response for the core/navigation block and adds the first and last inner blocks.
+ *
+ * @param WP_REST_Response $response The response object.
+ * @param WP_Post          $post     Post object.
+ * @param WP_REST_Request  $request  Request object.
+ * @return WP_REST_Response $response The response object.
+ */
+function gutenberg_hook_blocks_into_core_navigation_rest_response( $response, $post ) {
+	if ( ! isset( $response->data['content']['raw'] ) || ! isset( $response->data['content']['rendered'] ) ) {
+		return $response;
+	}
+	$parsed_blocks                         = parse_blocks( $response->data['content']['raw'] );
+	$content                               = serialize_blocks( gutenberg_hook_first_last_children_into_core_navigation( $parsed_blocks, $post->ID ) );
+	$response->data['content']['raw']      = $content;
+	$response->data['content']['rendered'] = apply_filters( 'the_content', $content );
+
+	return $response;
+}
+
+/**
+ * Hooks into the core/navigation adds the first_child and last_child hooked blocks during the server-side render process.
+ *
+ * @param WP_Block_List $inner_blocks The inner blocks of the core/navigation block.
+ * @param int|null $post_id Post ID of the navigation block.
+ * @return WP_Block_List $inner_blocks with the first and last inner blocks added.
+ */
+function gutenberg_hook_blocks_into_core_navigation( $inner_blocks, $post_id ) {
+	if ( ! $inner_blocks instanceof WP_Block_List ) {
+		return $inner_blocks;
+	}
+
+	$parsed_blocks = array_map(
+		function ( $inner_block ) {
+			return $inner_block->parsed_block;
+		},
+		iterator_to_array( $inner_blocks )
+	);
+	return new WP_Block_List( gutenberg_hook_first_last_children_into_core_navigation( $parsed_blocks, $post_id ) );
+}
+
+/**
+ * Adds the first and last inner blocks if there are hooked blocks.
+ *
+ * @param array $inner_blocks Parsed inner blocks of the core/navigation block.
+ * @param int $post_id Post ID of the navigation block.
+ * @return array $inner_blocks Parsed inner blocks maybe with hooked blocks.
```

Comment:
```suggestion
 * @return array Parsed inner blocks maybe with hooked blocks.
```

Reply:
Fixed in https://github.com/WordPress/gutenberg/pull/57754/commits/30dcb6470410687e25220113c13e1eb0d187435d